### ISSUE TITLE
ENH: define np.tau

### ISSUE THREAD

**Issue by hhoppe** [numpy/numpy#22153]
### Proposed new feature or change:

I really admire that Python 3.6 defines `math.tau` as the mathematical constant τ = 6.283185…:

> To learn more about Tau, check out Vi Hart’s video [Pi is (still) Wrong](https://www.youtube.com/watch?v=jG7vhMMXagQ), and start celebrating [Tau day](https://tauday.com/) by eating twice as much pie!

The `math` package defines only 5 constants: `pi`, `e`, `tau`, `inf`, and `nan`.
`numpy` defines [more constants](https://numpy.org/doc/stable/reference/constants.html) but omits `tau`.
Please include it too.  (Often we import `numpy` but not `math`.)

**Comment by WarrenWeckesser** [comment]
This was considered years ago, and it was decided at that time to not include `tau` in NumPy; see https://github.com/numpy/numpy/pull/9696

**Comment by jrmoserbaltimore** [comment]
> This was considered years ago, and it was decided at that time to not include `tau` in NumPy; see #9696

This kind of answer is ridiculous.  We made a lot of decisions at the time that in hindsight were ridiculous.  All of human history is basically horrible tyranny and oppression until people eventually decide prior policy decisions were bad.  Saying "we decided this a few years ago" without exploring why the decision stands is just ignorant and disrespectful.
